sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -180,13 +180,14 @@ object Cast {
 
     case (FloatType | DoubleType, TimestampType) => true
     case (TimestampType, DateType) => false
+    case (TimestampType, _: IntegralType) if to != LongType => true
```

Comment:
Is this related to this pr?

Reply:
Yes, converting TimeStamp to Byte/Short/Int type can be null